key-manager/src/main/scala/org/bitcoins/keymanager/EncryptedMnemonic.scala

```diff
@@ -9,9 +9,12 @@ import scodec.bits.ByteVector
 
 import scala.util.{Failure, Success, Try}
 
-case class DecryptedMnemonic(
-    mnemonicCode: MnemonicCode,
-    creationTime: Instant) {
+sealed trait WritableMnemonic {
```

Comment:
nit: Maybe rename this to `MnemonicState` ? I agree that both `DecryptedMnemonic` and `EncryptedMnemonic` are writable, but I think this class is more powerful -- it represents the state of the Mnemonic too. 

I.e. you can decrypt the `EncryptedMnemonic` to get a `DecryptedMnemonic` too presumably?